File: 1-ioc_di/src/main/java/it/italiancoders/demodi/DemoDiApplication.java
package it.italiancoders.demodi;

import it.italiancoders.demodi.components.ReportExec;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.context.ApplicationListener;
import org.springframework.context.annotation.Bean;

import java.util.ArrayList;
import java.util.List;

@SpringBootApplication
public class DemoDiApplication implements ApplicationListener<ApplicationReadyEvent> {

	@Autowired
	ReportExec reportExec;

	public static void main(String[] args) {
		SpringApplication.run(DemoDiApplication.class, args);

	}

	@Bean
	List<String> studentList(){
		List<String> students = new ArrayList<>();
		students.add("pippo");
		students.add("pluto");

		return students;
	}

	@Override
	public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
		reportExec.execute();

	}
}

File: 1-ioc_di-kotlin/src/main/kotlin/it/italiancoders/demodi/DemoDiApplication.kt
package it.italiancoders.demodi

import it.italiancoders.demodi.components.ReportExec
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.event.ApplicationReadyEvent
import org.springframework.context.ApplicationListener
import org.springframework.context.annotation.Bean

@SpringBootApplication
class DemoDiApplication : ApplicationListener<ApplicationReadyEvent> {
    @Bean
    fun studentsList(): List<String> = arrayListOf("pippo", "pluto")

    @Autowired lateinit var reportExec: ReportExec

    override fun onApplicationEvent(p0: ApplicationReadyEvent?) {
        reportExec.execute()
    }
}

fun main(args: Array<String>) {
    SpringApplication.run(DemoDiApplication::class.java, *args)
}
